commons/src/test/java/com/fincity/saas/commons/util/CodeUtilTest.java

```diff
@@ -1,16 +1,161 @@
 package com.fincity.saas.commons.util;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
 import org.junit.jupiter.api.Test;
 
 class CodeUtilTest {
 
-    @Test
-    void test() {
+	CodeUtil.CodeGenerationConfiguration cd = new CodeUtil.CodeGenerationConfiguration();
+
+	@Test
+	void test() {
+
+		cd.setSeparators(new int[] { 4, 6 });
+		String st = CodeUtil.generate(cd);
+
+		assertEquals(10, st.length());
+		assertEquals('-', st.charAt(4));
+		assertEquals('-', st.charAt(7));
+		assertTrue(st.matches("^[0-9-]+$"));
```

Comment:
Please change the regex expression as it is only covers whether only starting character but not covers overall length of the code which was generated. for eg: it will return true for any length of the string if any a-z or - condition matches.